core/templates/dev/head/services/PlaythroughServiceSpec.js

```diff
@@ -16,104 +16,93 @@
  */
 
-describe('Playthrough service', function() {
+describe('PlaythroughService', function() {
   beforeEach(module('oppia'));
   beforeEach(inject(function($injector) {
     this.PlaythroughService = $injector.get('PlaythroughService');
-    this.PlaythroughIssuesService = $injector.get('PlaythroughIssuesService');
     this.LearnerActionObjectFactory =
       $injector.get('LearnerActionObjectFactory');
-
-    this.whitelistedExpId = 'expId1';
-    this.blacklistedExpId = 'expId2';
-    this.expVersion = 1;
+    this.ExplorationFeaturesService =
+      $injector.get('ExplorationFeaturesService');
   }));
 
-  describe('Test playthrough service functions', function() {
-    beforeEach(function() {
-      this.PlaythroughIssuesService.initSession(
-        this.whitelistedExpId, this.expVersion, [this.whitelistedExpId]);
+  describe('.initSession', function() {
+    it('stores the correct values', function() {
+      this.expId = 'expId';
+      this.expVersion = 1;
+      this.playthroughRecordingProbability = 1.0;
+
       this.PlaythroughService.initSession(
-        this.whitelistedExpId, this.expVersion, 1.0);
-    });
+        this.expId, this.expVersion, this.playthroughRecordingProbability);
 
-    it('should initialize a session with correct values.', function() {
       var playthrough = this.PlaythroughService.getPlaythrough();
-
-      expect(playthrough.whitelistedExpId, this.whitelistedExpId);
-      expect(playthrough.expVersion, this.expVersion);
+      expect(playthrough.expId).toEqual(this.expId);
+      expect(playthrough.expVersion).toEqual(this.expVersion);
+      expect(playthrough.actions).toEqual([]);
+      expect(playthrough.issueCustomizationArgs).toEqual({});
     });
+  });
 
-    it('should record exploration start action.', function() {
-      this.PlaythroughService.recordExplorationStartAction('initStateName1');
-      var playthrough = this.PlaythroughService.getPlaythrough();
+  describe('recording exploration playthroughs', function() {
+    beforeEach(function() {
+      this.expId = 'expId';
+      this.expVersion = 1;
+      this.playthroughRecordingProbability = 1.0;
 
-      expect(playthrough.actions).toEqual(
-        [this.LearnerActionObjectFactory.createNew(
-          'ExplorationStart', {
-            state_name: {
-              value: 'initStateName1'
-            }
-          }, 1
-        )]);
+      this.PlaythroughService.initSession(
+        this.expId, this.expVersion, this.playthroughRecordingProbability);
+      spyOn(this.ExplorationFeaturesService, 'isPlaythroughRecordingEnabled')
+        .and.returnValue(true);
     });
 
-    it('should record answer submit action.', function() {
-      this.PlaythroughService.recordAnswerSubmitAction(
-        'stateName1', 'stateName2', 'TextInput', 'Hello', 'Try again', 30);
-      var playthrough = this.PlaythroughService.getPlaythrough();
+    describe('.recordExplorationStartAction', function() {
```

Comment:
Maybe add parentheses? I lean towards thinking that .name is a variable and .name() is a function... Maybe it is a personal preference, but if you feel it improves readability without parentheses, feel free to ignore this comment, and the one above. (Also, if you do add them, please add them at all locations in this file)

Reply:
Done.